### PR TITLE
Add new code style rules (.NET 8)

### PR DESCRIPTION
**Pull request by RoosterDragon**
Add new code style rules from .NET 8. We don't enforce them yet since we support a .NET 6 target, but we can still get them into the config and fix them up. Also take the chance to fix violations for some .NET 7 rules, again without enforcing the rule as yet. (See #21011)

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -724,6 +728,9 @@ dotnet_diagnostic.CA1417.severity = warning
 # Use 'nameof' in place of string.
 dotnet_diagnostic.CA1507.severity = warning
 
+# Avoid redundant length argument.
+dotnet_diagnostic.CA1514.severity = warning # TODO: Change to warning once using .NET 8 or later.
```
Comment (pchote) [inline .editorconfig]: The comments imply that these should be something other than warning?

Comment (RoosterDragon) [reply .editorconfig]: Indeed I forgot to actually downgrade them. Duh. Changed to `suggestion` throughout.

**Review by PunkPun — APPROVED** [review]
overall LGTM, untested

**Review by penev92 — APPROVED** [review]
LGTM, partially tested

**Comment by penev92** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/b5186bcc5939af2620b9c1e6774bc465e8c55ccb...d7a59cc0bf2dbc097831fbb7e1e4652102dbdcaa)

**Pull request merged**